enclave-runtime/src/lib.rs

```diff
@@ -611,6 +592,55 @@ where
 	Ok(())
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn sync_parentchain(
+	blocks_to_sync: *const u8,
+	blocks_to_sync_size: usize,
+	nonce: *const u32,
+) -> sgx_status_t {
+	let blocks_to_sync = match Vec::<SignedBlock>::decode_raw(blocks_to_sync, blocks_to_sync_size) {
+		Ok(blocks) => blocks,
+		Err(e) => return Error::Codec(e).into(),
+	};
+
+	if let Err(e) = sync_parentchain_int::<Block>(blocks_to_sync, *nonce) {
+		return e.into()
+	}
+
+	sgx_status_t::SGX_SUCCESS
+}
+
+/// Internal [`sync_parentchain`] function to be able to use the handy `?` operator.
+///
+/// Sync parentchain blocks to the light-client
+///
+/// This function makes an ocall that does the following:
+///
+/// *   send `confirm_call` xt's of the `Stf` functions executed due to in-/direct invocation to the
+///     to the parentchain
+/// *   sends sidechain `confirm_block` xt's with the produced sidechain blocks
+/// *   gossip produced sidechain blocks to peer validateers.
+fn sync_parentchain_int<PB>(blocks_to_sync: Vec<SignedBlockG<PB>>, _nonce: u32) -> Result<()>
```

Comment:
you renamed the block execution function explicitely to `execute_trusted_operations_internal`, is there a reason you're only using `_int` here?

If not, I'd like to request to decide for one naming convention - this will make it much easier to explain to other developers.